node/pkg/watchers/evm/chain_config.go

```diff
@@ -0,0 +1,274 @@
+
+// NOTE: Whenever changes are made to the config data in this file, you should do the following:
+//    node/pkg/watcher/evm$ go test
+//    node/pkg/watcher/evm/verify_chain_config$ go run verify.go
+
+// TODO: In a future PR we could consider merging the data in `node/hack/repair_eth/repair_eth.go` into here.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+type (
+	// EnvEntry specifies the config data for a given chain / environment.
+	EnvEntry struct {
+		// Finalized indicates if the chain supports querying for finalized blocks.
+		Finalized bool
+
+		// Safe indicates if the chain supports querying for safe blocks.
+		Safe bool
+
+		// EvmChainID is the expected EVM chain ID (what is returned by the `eth_chainId` RPC call).
+		EvmChainID uint64
+
+		// PublicRPC is not actually used by the watcher. It's used to verify that the EvmChainID specified here is correct.
+		PublicRPC string
+	}
+
+	// EnvMap defines the config data for a given environment (mainet or testnet).
+	EnvMap map[vaa.ChainID]EnvEntry
+)
+
+var (
+	ErrInvalidEnv = errors.New("invalid environment")
+	ErrNotFound   = errors.New("not found")
+
+	// mainnetChainConfig specifies the configuration for all chains enabled in Mainnet.
+	// NOTE: Only add a chain here if the watcher should allow it in Mainnet!
+	// NOTE: If you change this data, be sure and run the tests described at the top of this file!
+	mainnetChainConfig = EnvMap{
+		vaa.ChainIDEthereum: {Finalized: true, Safe: true, EvmChainID: 1, PublicRPC: "https://ethereum-rpc.publicnode.com"},
+		vaa.ChainIDBSC:      {Finalized: true, Safe: true, EvmChainID: 56, PublicRPC: "https://bsc-rpc.publicnode.com"},
+
+		// Polygon supports polling for finalized but not safe: https://forum.polygon.technology/t/optimizing-decentralized-apps-ux-with-milestones-a-significantly-accelerated-finality-solution/13154
+		vaa.ChainIDPolygon: {Finalized: true, Safe: false, EvmChainID: 137, PublicRPC: "https://polygon-bor-rpc.publicnode.com"},
+
+		vaa.ChainIDAvalanche: {Finalized: false, Safe: false, EvmChainID: 43114, PublicRPC: "https://avalanche-c-chain-rpc.publicnode.com"},
+		vaa.ChainIDOasis:     {Finalized: false, Safe: false, EvmChainID: 42262, PublicRPC: "https://emerald.oasis.dev/"},
+		// vaa.ChainIDAurora:    Not supported in the guardian.
+		vaa.ChainIDFantom:   {Finalized: false, Safe: false, EvmChainID: 250, PublicRPC: "https://fantom-rpc.publicnode.com"},
+		vaa.ChainIDKarura:   {Finalized: true, Safe: true, EvmChainID: 686, PublicRPC: "https://eth-rpc-karura.aca-api.network/"},
+		vaa.ChainIDAcala:    {Finalized: true, Safe: true, EvmChainID: 787, PublicRPC: "https://eth-rpc-acala.aca-api.network/"},
+		vaa.ChainIDKlaytn:   {Finalized: false, Safe: false, EvmChainID: 8217}, // As of Feb 2025, can't find a working public node.
```

Comment:
KLAYTN_RPC="https://rpc.ankr.com/klaytn"

Reply:
This endpoint also gives `ERROR: Failed to query EVM chain ID for test klaytn: failed to read evm chain id: API key is not allowed to access blockchain`.